custom_components/localtuya/pytuya/__init__.py

```diff
@@ -487,6 +503,15 @@ async def heartbeat(self):
         """Send a heartbeat message."""
         return await self.exchange(HEARTBEAT)
 
+    async def reset(self):
```

Comment:
This broadly makes sense. The version of pytuya here has been actively maintained in other projects like https://github.com/jasonacox/tinytuya/blob/master/tinytuya/__init__.py - did you want to take a quick look and see if they've solved similar issues; and if so, is there anything you'd backport here?

Reply:
Thanks for the feedback! I took a brief look at the linked code but it seems quite different.

I’ll look at updating this branch with the v4 updates and ensure it all works